### PR TITLE
SW-18384 Added guide on how to use eslint in shopware plugins

### PR DESCRIPTION
**Pull request by klarstil**
Ping @ngoedde @SimonBaeumer

### CODE REVIEW

File: source/designers-guide/eslint-for-plugins/index.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+menu_title: ESLint for plugins
+---
+
+### Introduction
+Shopware features an automatic static code analyses tool called [ESLint](http://eslint.org/). Combined with a shared configuration called [standardjs](https://standardjs.com/) we provided a convenient way to ensure a consistent code style in the core & in our plugins.
+
+In this guide we'll talking a look on how you as a Shopware plugin developer can use the benefits of ESLint in your own plugin.
+
+### Installation
+ESLint has Node.js and NPM as an dependency you have to install beforehand. If you haven't installed them, please go ahead and install it using the provided [download methods](https://nodejs.org/en/download/current/).
```
Comment (mitelg) [inline source/designers-guide/eslint-for-plugins/index.md]: ESLint has Node.js and NPM as dependencies, which you have to install beforehand.

File: source/designers-guide/eslint-for-plugins/index.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+Next we're installing ESLint and it's dependencies globally on your system. You have to install them once and be ready for every other plugin you'll write.
+
+Use the following command to install ESLint, the standardjs configuration & necessary plugins:
+
+```
+npm install -g eslint eslint-config-standard eslint-plugin-import eslint-plugin-node eslint-plugin-promise eslint-plugin-standard
+```
+
+#### Installating the dependencies per plugin
+If you're not feeling comfortable installing the dependencies globally, you can also create a `package.json` file in your plugin directory (e.g. `SwagBundle` for example). You can create this file using the command `npm init`.
```
Comment (mitelg) [inline source/designers-guide/eslint-for-plugins/index.md]: (e.g. `SwagBundle` for example) is double 

e.g. actually means for example :wink:

File: source/designers-guide/eslint-for-plugins/index.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+After the file was successfully created, run the following command to install the dependencies as `devDependencies`:
+
+```
+npm install --save-dev eslint eslint-config-standard eslint-plugin-import eslint-plugin-node eslint-plugin-promise eslint-plugin-standard
+```
+
+### Running ESLint
+Starting with [Shopware 5.2](http://community.shopware.com/Downloads_cat_448.html) we provided you a ESLint configuration file which can be found under `themes/.eslintrc.js`. We're using this file for our plugins as well to provide a consistent coding styling through the application.
```
Comment (mitelg) [inline source/designers-guide/eslint-for-plugins/index.md]: we should mention the right SW version here. afaik the eslintrc.js is not in 5.2.0

File: source/designers-guide/eslint-for-plugins/index.md (hunk 1 above)
Comment (klarstil) [inline source/designers-guide/eslint-for-plugins/index.md]: Done ✅

File: source/designers-guide/eslint-for-plugins/index.md (hunk 2 above)
Comment (klarstil) [inline source/designers-guide/eslint-for-plugins/index.md]: Done ✅

File: source/designers-guide/eslint-for-plugins/index.md (hunk 3 above)
Comment (klarstil) [inline source/designers-guide/eslint-for-plugins/index.md]: Done ✅

**Pull request merged**